ouroboros-consensus-cardano/test/Test/Consensus/Cardano/Examples.hs

```diff
@@ -67,37 +69,80 @@ import           Test.Consensus.Cardano.Generators (toTelescope')
 
 type Crypto = TPraosMockCrypto ShortHash
 
-eraExamples :: NP Golden.Examples (CardanoEras Crypto)
+eraExamples :: NP Examples (CardanoEras Crypto)
 eraExamples = Byron.examples :* Shelley.examples :* Nil
 
+-- | By using this function, we can't forget to update this test when adding a
+-- new era to 'CardanoEras'.
 combineEras ::
-     forall f byron shelley cardano.
-     ( Coercible byron   (f ByronBlock)
-     , Coercible shelley (f (ShelleyBlock Crypto))
-     , Coercible cardano (f (CardanoBlock Crypto))
-     )
-  => Proxy f
-  -> (forall blk. Golden.Examples blk -> NonEmpty (f blk))
-  -> (byron   -> cardano)
-  -> (shelley -> cardano)
-  -> NonEmpty cardano
-combineEras _ getExample injByron injShelley =
-    join $
-    hcollapseNE_NP $
-    hliftA2
-      aux
-      eraInjections
-      eraExamples
+     NP Examples (CardanoEras Crypto)
+  -> Examples (CardanoBlock Crypto)
+combineEras = mconcat . hcollapse . hap eraInjections
   where
-    eraInjections :: NP (f -.-> K (f (CardanoBlock Crypto))) (CardanoEras Crypto)
-    eraInjections = fn (K . coerce injByron) :* fn (K . coerce injShelley) :* Nil
-
-    aux ::
-         forall blk.
-         (f -.-> (K (f (CardanoBlock Crypto)))) blk
-      -> Golden.Examples blk
-      -> K (NonEmpty cardano) blk
-    aux inj exs = K (coerce . unK . apFn inj <$> getExample exs)
+    eraInjections :: NP (Examples -.-> K (Examples (CardanoBlock Crypto)))
+                        (CardanoEras Crypto)
+    eraInjections =
+           fn (K . injExamplesByron)
+        :* fn (K . injExamplesShelley)
+        :* Nil
+
+-- Generalise in #2368
+injExamplesByron :: Examples ByronBlock -> Examples (CardanoBlock Crypto)
```

Comment:
I now have two such injections functions instead of one generic one, because these two are much simpler (and not too wide :grin:). When we have #2368 we could replace them again by a single generic one.

I still use `combineEras`, so adding an era will break the tests.